apps/aechannel/src/aesc_fsm_min_depth_watcher.erl

```diff
@@ -0,0 +1,147 @@
+                                     min_depth => MinDepth}, [{debug,[trace]}]).
+
+init(#{} = Arg) ->
+    lager:debug("started min_depth watcher for ~p", [maps:get(fsm, Arg)]),
+    true = aec_events:subscribe(top_changed),
+    true = aec_events:subscribe(block_created),
+    lager:debug("subscribed to top_changed & block_created", []),
+    self() ! check_status,
+    {ok, Arg}.
+
+%% handle_info({gproc_ps_event, top_changed, #{info := Block}},
+%%             #{chan_id   := ChanId,
+%%               min_depth := MinDepth,
+%%               fsm       := Fsm} = St) ->
+    %% CurHeight = aec_blocks:height(Block),
+    %% lager:debug("got top_changed, block at height ~p", [CurHeight]),
+    %% case maps:find(channel_at, St) of
+    %%     {ok, ChHeight} when CurHeight - ChHeight >= MinDepth ->
+    %%         lager:debug("reached minimum height (~p - ~p >= ~p",
+    %%                     [CurHeight, ChHeight, MinDepth]),
+    %%         aesc_fsm:own_funding_locked(Fsm, ChanId),
+    %%         {stop, normal, St};
+    %%     {ok, ChHeight} when CurHeight - ChHeight < MinDepth ->
+    %%         lager:debug("not yet mininum height (~p - ~p < ~p)",
+    %%                     [CurHeight, ChHeight, MinDepth]),
+    %%         {noreply, St};
+    %%     error ->
+    %%         lager:debug("haven't yet found the tx", []),
+    %%         case block_has_channel(ChanId, Block) of
+    %%             true when MinDepth == 0 ->
+    %%                 lager:debug("block has channel tx, MinDepth = 0", []),
+    %%                 aesc_fsm:own_funding_locked(Fsm, ChanId),
+    %%                 {stop, normal, St};
+    %%             true ->
+    %%                 lager:debug("block has channel tx", []),
+    %%                 {noreply, St#{channel_at => CurHeight}};
+    %%             false ->
+    %%                 lager:debug("block doesn't have channel tx", []),
+    %%                 {noreply, St}
+    %%         end
+    %% end;
+handle_info({gproc_ps_event, top_changed, _}, St) ->
+    check_status(St);
+handle_info({gproc_ps_event, block_created, _}, St) ->
+    check_status(St);
+handle_info(check_status, St) ->
+    check_status(St);
+handle_info(_Msg, St) ->
+    lager:debug("got unknown Msg: ~p", [_Msg]),
+    {noreply, St, 1000}.
+
+
+handle_call(_Req, _From, St) ->
+    {reply, {error, unknown_request}, St}.
+
+handle_cast(_Msg, St) ->
+    {noreply, St}.
+
+terminate(_Reason, _St) ->
+    ok.
+
+code_change(_FromVsn, St, _Extra) ->
+    {ok, St}.
+
+check_status(#{fsm := Fsm, chan_id := ChanId,
+               tx_hash := TxHash, min_depth := MinDepth} = St) ->
+    lager:debug("check_status(~p)", [Fsm]),
+    case min_depth_achieved(TxHash, MinDepth) of
+        true ->
+            lager:debug("min_depth achieved", []),
+            aesc_fsm:own_funding_locked(Fsm, ChanId),
+            {stop, normal, St};
+        _ ->
+            lager:debug("min_depth not yet achieved", []),
+            {noreply, St, 5000}
```

Comment:
The timeout intervals are too strict, please remove them.

Reply:
That timeout is just for debugging purposes. There should not be a timeout at all, eventually. Also, the gen_server/gen_statem debug tracing shall be removed.